api/src/main/kotlin/se/ansman/kotshi/KotshiUtils.kt

```diff
@@ -23,28 +23,34 @@ object KotshiUtils {
             }.append(propertyName)
 
     @JvmStatic
-    fun <T : Annotation> createJsonQualifierImplementation(annotationType: Class<T>): T {
+    fun <T : Annotation> createJsonQualifierImplementation(
+            annotationType: Class<T>, elementNameToArg: Map<String, Any>
+    ): T {
         if (!annotationType.isAnnotation) {
             throw IllegalArgumentException("$annotationType must be an annotation.")
         }
         if (!annotationType.isAnnotationPresent(JsonQualifier::class.java)) {
             throw IllegalArgumentException("$annotationType must have @JsonQualifier.")
         }
-        if (annotationType.declaredMethods.isNotEmpty()) {
-            throw IllegalArgumentException("$annotationType must not declare methods.")
-        }
         @Suppress("UNCHECKED_CAST")
         return Proxy.newProxyInstance(annotationType.classLoader, arrayOf<Class<*>>(annotationType)) { proxy, method, args ->
             when (method.name) {
                 "annotationType" -> annotationType
                 "equals" -> annotationType.isInstance(args[0])
                 "hashCode" -> 0
                 "toString" -> "@${annotationType.name}()"
-                else -> method.invoke(proxy, *args)
+                else -> elementNameToArg[method.name] ?: method.invoke(proxy, args)
             }
         } as T
     }
 
+    @JvmStatic
+    fun putInMap(map: MutableMap<String, Any>, key: String, value: Any): MutableMap<String, Any> = map.apply {
```

Comment:
Make an extension instead? Perhaps with the name `putWithoutConflict` or something similar.